### ISSUE TITLE
Some metrics are gone after upgrading to 2023-04-07T05-28-58Z

### ISSUE THREAD

**Issue by JiffsMaverick** [minio/minio#17022]
Some metrics disappeared after upgrading MinIO from RELEASE.2023-03-24T21-41-23Z to RELEASE.2023-04-07T05-28-58Z.

List of problematic metrics from v2 (/minio/v2/metrics/cluster)
- minio_s3_requests_total
- minio_s3_requests_errors_total
- minio_s3_requests_5xx_errors_total
- minio_s3_requests_4xx_errors_total
- minio_cluster_disk_free_inodes
- minio_s3_time_ttfb_seconds_distribution

List of problematic metrics from v1 (/minio/prometheus/metrics):
- s3_requests_total
- s3_errors_total
- s3_ttfb_seconds_bucket

## Expected Behavior
I'm expecting to get the metrics.

## Current Behavior
I get "no data".

## Steps to Reproduce (for bugs)
You can check this problem on the official demo server with a command like:

```curl https://play.min.io/minio/v2/metrics/cluster | grep -i minio_s3_requests_total```

## Regression
Yes, there are no problems with RELEASE.2023-03-24T21-41-23Z.

## Your Environment
* Version used (`minio --version`): RELEASE.2023-04-07T05:28:58Z
* Server setup and configuration:
```
❯ mc admin info my-cluster/
●  x.x.x.x:9000
   Uptime: 22 hours
   Version: 2023-04-07T05:28:58Z
   Network: 4/4 OK
   Drives: 1/1 OK
   Pool: 1

●  x.x.x.x:9000
   Uptime: 22 hours
   Version: 2023-04-07T05:28:58Z
   Network: 4/4 OK
   Drives: 1/1 OK
   Pool: 1

●  x.x.x.x:9000
   Uptime: 22 hours
   Version: 2023-04-07T05:28:58Z
   Network: 4/4 OK
   Drives: 1/1 OK
   Pool: 1

●  x.x.x.x:9000
   Uptime: 22 hours
   Version: 2023-04-07T05:28:58Z
   Network: 4/4 OK
   Drives: 1/1 OK
   Pool: 1

Pools:
   1st, Erasure sets: 1, Drives per erasure set: 4

1 GiB Used, 1 Bucket, 37 Objects
4 drives online, 0 drives offline
```
* Operating System and version (`uname -a`): a cluster running with the official Docker image

**Comment by harshavardhana** [comment]
This bug is mostly via 

```
git rel RELEASE.2023-03-24T21-41-23Z..master cmd/http*
- Use one http response recorder per external http call (#16938) (03/31/23) (Anis Eleuch)
```

@vadmeste this is the issue

**Comment by MalKeshar** [comment]
@harshavardhana, I still don't see metric minio_s3_time_ttfb_seconds_distribution in RELEASE.2023-05-18T00-05-36Z and RELEASE.2023-05-27T05-56-19Z. Should I open separate issue?

**Comment by chthtlo** [comment]
@harshavardhana, Hello, I have the same issue when updated from 2023-04-20T17:56:55Z to 2023-11-11T08-14-41Z, don't see few metrics:
```
minio_s3_requests_errors_total
minio_s3_requests_5xx_errors_total
minio_s3_requests_4xx_errors_total
```
Can you look at that please?

**Comment by tariver** [comment]
I have 2023-11-11T08-14-41Z and don't see these metrics either, although [the official dashboard](https://github.com/minio/minio/blob/b3314e97a64a42d22ba5b939917939d72a28c97d/docs/metrics/prometheus/grafana/minio-dashboard.json) uses some of them.

**Comment by tariver** [comment]
OK, I've done some testing, turns out that this metrics _appear_ only after some operation with S3 (like uploading a file) is done.
